### PR TITLE
Graphs from OT

### LINKED ISSUE #2: Extend ndim > 2 and contours
Use `ot.Pairs` if ndim > 2 and with ndim = 2 change mpl with `ot.Contour`.

### PR DESCRIPTION
**Pull request by tupui**
Replace matplotlib graphs with OT.

Closes #2

### CODE REVIEW

**Comment by tupui** [comment]
I will add another commit to replace other mpl graphs.

**Comment by tupui** [comment]
The graphs are now done using OT except for the `fill_between` which is still some mpl. It seems that we could do that using `ot.Polygon`. What do you think?

**Comment by tupui** [comment]
I also made an attempt for the computation of the mean using the distribution. I am not sure though about the method. I simply set a very low threshold for the computation of `computeMinimumVolumeLevelSetWithThreshold` and I use the corresponding sample.

**Comment by tupui** [comment]
As a side note, the API for the Graphs is not consistent. For Contour, Pairs or Cloud the order of common arguments is not the same.

**Comment by mbaudin47** [comment]
> I also made an attempt for the computation of the mean using the distribution. The mean is found as the curve which produce the max PDF value.
> @mbaudin47 Do we need to try to do an optimization to find the mean based on the KDE? If yes, using which optimization method from OT?

My plan for the ProcessHDRAlgo is to first separate the core of the estimation of the density algorithm from the code so that we can use an alternative to KDE, e.g. a gaussian mixture. Afterwards, we might better see which option can be used for *any* density estimate, rather than focus on potential properties of the KDE.
On your specific question, my guess is that any optimizer should do the trick, but a gradient based method should perform best: this should be easy in theory, given that a density has its gradients and that KDE is by construction a regular function. My best option would be NLOPT/LD_LBFGS, but the doc seems to require bounds on the parameters, which is not appropriate here: this seems to require more experiments.

File: othdrplot/process_high_density_region_algorithm.py [hunk 1]
```diff
@@ -157,44 +158,58 @@ def plotTrajectories(self):
             graph.add(traj_curves)
 
         # Plot mean
-        mean_field = self.processSample.computeMean()
-        mean_curve = ot.Curve(t, mean_field.getValues(), 'Discrete mean')
+        if discreteMean:
```
Comment (mbaudin47) [inline othdrplot/process_high_density_region_algorithm.py]: The code is fine, but I would rather call it "central_field", given that in the case that this is the "mode", the name "mean_field" might be misleading. It does not prevent from pushing the code anyway.

File: othdrplot/process_high_density_region_algorithm.py [hunk 2]
```diff
@@ -157,44 +158,58 @@ def plotTrajectories(self):
...
+        if plotInliers:
             ax.plot(t, inlierSample, "b-")
 
         # Plot inlier bounds
-        inlierMin = np.min(inlierSample, axis=1)
+        inlier_min = np.min(inlierSample, axis=1)
         inlierMax = np.max(inlierSample, axis=1)
-        ax.fill_between(t, inlierMin, inlierMax, where=inlierMax >= inlierMin,
-                        facecolor='green', label="Inlier at alpha=%.4f" % (self.outlierAlpha))
+        ax.fill_between(t, inlier_min, inlierMax, facecolor='green',
```
Comment (mbaudin47) [inline othdrplot/process_high_density_region_algorithm.py]: This is the place where Polygon and PolygonArray come into play.

**Comment by mbaudin47** [comment]
Here is a subplot example which might be interesting in order to be combined with a Contour in the general case where dim>2.

```
import openturns as ot
import pylab as pl

myBeta = ot.Beta(5, 7, 9, 10)
myPDFBeta = myBeta.drawPDF()
myCDFBeta = myBeta.drawCDF()
myExponential = ot.Exponential(3)
myPDFExp = myExponential.drawPDF()
myCDFExp = myExponential.drawCDF()


fig = pl.figure(figsize=(12, 4))
ax_pdf = fig.add_subplot(2, 2, 1)
_ = ot.viewer.View(myPDFBeta, figure=fig, axes=[ax_pdf])
ax_cdf = fig.add_subplot(2, 2, 2)
_ = ot.viewer.View(myCDFBeta, figure=fig, axes=[ax_cdf])
ax_pdf_exp = fig.add_subplot(2, 2, 3)
_ = ot.viewer.View(myPDFExp, figure=fig, axes=[ax_pdf_exp])
ax_cdf_exp = fig.add_subplot(2, 2, 4)
_ = ot.viewer.View(myCDFExp, figure=fig, axes=[ax_cdf_exp])
```

image

**Pull request merged**